lib/vector-core/src/event/metadata.rs

```diff
@@ -70,6 +73,16 @@ impl EventMetadata {
         &mut self.secrets
     }
 
+    /// Returns a mutable reference to the metadata source.
+    pub fn source(&self) -> &Option<Arc<OutputId>> {
+        &self.source
+    }
+
+    /// Returns a mutable reference to the metadata source.
+    pub fn source_mut(&mut self) -> &mut Option<Arc<OutputId>> {
+        &mut self.source
+    }
```

Comment:
_Every_ use of this function takes the form `*x.source_mut() = Some(value);`. Would it be better to change the interface to a setter?
```suggestion
    /// Returns a mutable reference to the metadata source.
    pub fn set_source(&mut self, source_id: Arc<OutputId>) {
        self.source = Some(source_id);
    }
```

Reply:
Good point. A source will never be changed once set.

Reply:
If it will not be changed once set, is there some pattern that will prevent changing it, like a once cell?